test/vanilla/Expected/AcceptanceTests/ExtensibleEnums/Models/DaysOfWeekExtensibleEnum.cs

```diff
@@ -0,0 +1,115 @@
+// <auto-generated>
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+//
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+// </auto-generated>
+
+namespace Fixtures.AcceptanceTestsExtensibleEnums.Models
+{
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Defines values for DaysOfWeekExtensibleEnum.
+    /// </summary>
+    /// <summary>
+    /// Determine base value for a given allowed value if exists, else return
+    /// the value itself
+    /// </summary>
+    [JsonConverter(typeof(DaysOfWeekExtensibleEnumConverter))]
+    public struct DaysOfWeekExtensibleEnum : System.IEquatable<DaysOfWeekExtensibleEnum>
+    {
+        private DaysOfWeekExtensibleEnum(string underlyingValue)
+        {
+            UnderlyingValue=underlyingValue;
+        }
+
+        public static readonly DaysOfWeekExtensibleEnum Monday = "Monday";
+
+        public static readonly DaysOfWeekExtensibleEnum Tuesday = "Tuesday";
+
+        public static readonly DaysOfWeekExtensibleEnum Wednesday = "Wednesday";
+
+        public static readonly DaysOfWeekExtensibleEnum Thursday = "Thursday";
+
+        public static readonly DaysOfWeekExtensibleEnum Friday = "Friday";
+
+        public static readonly DaysOfWeekExtensibleEnum Saturday = "Saturday";
+
+        public static readonly DaysOfWeekExtensibleEnum Sunday = "Sunday";
+
+
+        /// <summary>
+        /// Underlying value of enum DaysOfWeekExtensibleEnum
+        /// </summary>
+        private readonly string UnderlyingValue;
+
+        /// <summary>
+        /// Returns string representation for DaysOfWeekExtensibleEnum
+        /// </summary>
+        public override string ToString()
+        {
+            return UnderlyingValue.ToString();
+        }
+
+        /// <summary>
+        /// Compares enums of type DaysOfWeekExtensibleEnum
+        /// </summary>
+        public bool Equals(DaysOfWeekExtensibleEnum e)
+        {
+            return UnderlyingValue.Equals(e.UnderlyingValue);
+        }
+
+        /// <summary>
+        /// Implicit operator to convert string to DaysOfWeekExtensibleEnum
+        /// </summary>
+        public static implicit operator DaysOfWeekExtensibleEnum(string value)
+        {
+            return new DaysOfWeekExtensibleEnum(value);
+        }
+
+        /// <summary>
+        /// Implicit operator to convert DaysOfWeekExtensibleEnum to string
+        /// </summary>
+        public static implicit operator string(DaysOfWeekExtensibleEnum e)
```

Comment:
This should probably be explicit, not implicit. Otherwise type safety is easily compromised, defeating the purpose of having this type in the first place.

Reply:
This was an ask from the dotnet sdk team; also this allows a natural transition for `modelAstString:true` in the future.  
@shahabhijeet @markcowl if you can verify this is okay, I can merge this ASAP

Reply:
@brjohnstmsft  -- From my perspective, I would have went with the explicit operator, but I'm not super passionate one way or the other.  The SDK team did make the point that it was a no-op code change for those who had been using strings.

Reply:
@fearthecowboy If folks don't care about type safety, then they can keep using strings. This is a new feature of AutoRest; IMO it's more than reasonable to require a few code changes to start using it, especially when it's as simple (and clear) as adding some explicit casts.

I'm curious to hear what @shahabhijeet and @markcowl have to say.

Reply:
What I know from hallway conversation was that, implicit conversion requirements are coming from PowerShell perspective. Mark will be right person to comment on the necessity to have support for implicit conversion.

Reply:
Perhaps we should make implicit/explicit optional.

Reply:
@fearthecowboy Sounds good to me.